static/js/compose.js

```diff
@@ -524,6 +530,21 @@ export function initialize() {
         },
     );
 
+    $("#compose_pm_mentions_user_alert").on(
+        "click",
+        ".compose_pm_mentions_user_alert_close",
+        (event) => {
+            const stream_alert_row = $(event.target).parents(".compose_pm_mentions_user_alert");
+            const stream_alert = $("#compose_pm_mentions_user_alert");
+
+            stream_alert_row.remove();
+
+            if (stream_alert.children().length === 0) {
+                stream_alert.hide();
+            }
+        },
+    );
```

Comment:
We can deduplicate some code here and use same handler for close buttons in all type of banners. That can probably be a follow-up.